src/format.rs

```diff
@@ -208,13 +208,28 @@ pub(crate) struct FormatProcessData<'a> {
     pub(crate) with_target: bool,
     #[cfg(feature = "ansi")]
     pub(crate) ansi: bool,
+    pub(crate) with_trimmed_directory: bool,
 }
 
 impl<'a> fmt::Display for FormatProcessData<'a> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         let thread_name = self.thread_name;
         let target = self.metadata.target();
-        let file = self.metadata.file().unwrap_or("");
+        let file = self
+            .metadata
+            .file()
+            .map(|f| {
+                if self.with_trimmed_directory {
+                    let path = Path::new(f);
+                    path.file_name()
+                        .map(OsStr::to_str)
+                        .unwrap_or_default()
+                        .unwrap_or_default()
```

Comment:
Wouldn't it be better to use the original full filename if something goes wrong here?

Reply:
Yes that is better. I added a fixup that does this.